File: backEnd/src/main/java/com/smart/safais/model/Payment.java
package com.smart.safais.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "booking_id")
    private Booking booking;

    private BigDecimal amount;
    private String paymentMethod;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Status status = Status.PENDING;

    @Column(name = "paid_at")
    private LocalDateTime paidAt;

    public enum Status {
        PENDING, PAID, FAILED
    }
} 

File: frontEnd/src/app/model/Payment.ts
import { Booking } from './Booking';

export interface Payment {
  id?: number;
  booking: Booking;
  amount: number;
  paymentMethod: string;
  status: 'PENDING' | 'PAID' | 'FAILED';
  paidAt?: string;
}

export interface PaymentResponse {
  success: boolean;
  message: string;
  data: Payment[];
} 